apps/admin-portal/src/pages/customize.tsx

```diff
@@ -0,0 +1,366 @@
+     * @param {string} initialValue - Default value set in the default theme less.
+     * @param {string} input - Input name.
+     * @param {string} [type] - Input type. E.g. "color"
+     * 
+     * @returns {inputProps}
+     */
+    const useStyleInput = (initialValue: string, input: string, type?: string) => {
+
+        const [ value, setValue ] = useState(initialValue);
+
+        const handleChange = (e) => {
+            setValue(e.target.value);
+        };
+
+        const inputProps = {
+            name: input,
+            onChange: handleChange,
+            placeholder: ("E.g. " + initialValue),
+            value
+        };
+
+        const handleColorPickerOnChange = (hexValue: string) => {
+            setValue(hexValue);
+            setThemeOptions({...themeOptions, [`@${input}`]: value });
+        };
+
+        const colorChildren = () => {
+            return (
+                <>
+                    <ColorPicker
+                        name={ input }
+                        placeholder={ "E.g. " + initialValue }
+                        inputColor={ value }
+                        inputOnChangeHandler={ handleColorPickerOnChange }
+                    />
+                </>
+            );
+        };
+
+        if (type !== undefined && type === "color") {
+            return {
+                children: colorChildren()
+            };
+        }
+
+        return inputProps;
+    };
+    
+    const primaryColor = useStyleInput(defaultThemeVariables["primaryColor"], "primaryColor", "color");
+    const pageBackground = useStyleInput(defaultThemeVariables["pageBackground"], "pageBackground", "color");
+
+    const handleCompileTheme = () => {
+        compile(themeOptions);
+    };
+
+    return (
+        <Form>
+            <Form.Field>
+                <label>App Primary Color</label>
+                <Grid columns={ 12 }>
+                    <Grid.Row>
+                        <Grid.Column>
+                            <Form.Input { ...primaryColor } />
+                        </Grid.Column>
+                    </Grid.Row>
+                </Grid>
+            </Form.Field>
+            <Form.Field>
+                <label>App Background Color</label>
+                <Grid columns={ 12 }>
+                    <Grid.Row>
+                        <Grid.Column>
+                            <Form.Input { ...pageBackground } />
+                        </Grid.Column>
+                    </Grid.Row>
+                </Grid>
+            </Form.Field>
+
+            <Button onClick={ () => handleCompileTheme() }>Update</Button>
+        </Form>
+    );
+};
+
+/**
+ * Customize Page.
+ *
+ * @return {ReactElement}
+ */
+export const CustomizePage = (): ReactElement => {
+
+    const { setAppName, setCopyrightText, setLogo, setProductName, setTheme, state } = useContext(ThemeContext);
+
+    const handleThemeSelect = (theme) => {
+        setTheme(theme);
+    };
+
+    const handleProductNameChange = (e) => {
```

Comment:
Would be better if we could access value like this IMO. So we don't need to have `e.target.value`.

```suggestion
    const handleProductNameChange = (e: ChangeEvent, { value }: { value: string }) => {
```

Reply:
Fixed in other places. But here we had to stick with this.